### DISCUSSION TITLE
"unauthorized" error when pushing images

### DISCUSSION THREAD

**Discussion by alpeb** [community/community#26815] [Packages]
<p>Hi,</p>
<p>I have succeeded on pushing images to <a href="http://ghcr.io" rel="nofollow noopener">ghcr.io</a> under an org when logged-in as an org owner, but when logged-in as just an org member <code>docker push</code> just returns <code>unauthorized</code>.</p>
<p>This org’s setting <code>Member privileges -&gt; Package creation</code> is already set to <code>Public</code>, and the PAT I’m using already has the <code>write:packages</code> permission.</p>
<p>Any idea what could be going on?</p>
<p>Thanks!</p>

**Comment by clarkbw** [comment]
<p>Hello!</p>
<p>Have you <a href="https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/configuring-access-control-and-visibility-for-container-images">configured access</a> for the organization member? The owner who created the Container would have admin permissions by default. However the org member would need to have <code>write</code> access added through the container settings.</p>
<p>Lemme know if that works!</p>

**Comment by alpeb** [comment, marked as answer]
<p>Thanks for the feedback <a class="mention" href="/clarkbw">@clarkbw</a>, that worked :+1:</p>
<p>So, if I understood correctly, only org owners can <em>create</em> images, and then they should grant access to other members in order for them to be able to push other tags for those images.</p>
<p>If I got that correct, then the description under the org’s <code>Member privileges -&gt; Package creation</code> “Members will be able to create public packages, visible to anyone” can be a little misleading :wink:</p>
<p>Thanks again!</p>

**Comment by clarkbw** [comment]
<aside class="quote no-group" data-username="alpeb" data-post="3" data-topic="131217">
<div class="title">
<div class="quote-controls"></div>
 alpeb:</div>
<blockquote>
<p>only org owners can <em>create</em> images, and then they should grant access to other members in order for them to be able to push other tags for those images.</p>
</blockquote>
</aside>
<p>Any org member can create an image. That creator holds admin for the image name and can extend it to others. Sorry I was describing your circumstance.</p>

**Comment by ataylorme** [comment]
<aside class="quote group-github-employees" data-username="clarkbw" data-post="4" data-topic="131217">
<div class="title">
<div class="quote-controls"></div>
 clarkbw:</div>
<blockquote>
<p>Any org member can create an image. That creator holds admin for the image name and can extend it to others. Sorry I was describing your circumstance.</p>
</blockquote>
</aside>
<p>I am seeing unauthorized when publishing a new container to  <code>ghcr.io/org-a</code>, which I am a member of. The same personal access token, with <code>write:packages</code>, is authorized and able to publish for <code>ghcr.io/org-b</code>, which I am an owner of.</p>
<p>I have looked and the documentation and forums and I see lots of information on how to adjust permissions for already created images.</p>
<p><a class="mention" href="/clarkbw">@clarkbw</a> can you clarify how an org member (non-owner) can push/create new images to <code>ghcr.io</code> in the org namespace?</p>

**Comment by clarkbw** [comment]
<aside class="quote no-group" data-username="ataylorme" data-post="5" data-topic="131217">
<div class="title">
<div class="quote-controls"></div>
 ataylorme:</div>
<blockquote>
<p>I am seeing unauthorized when publishing a new container to <code>ghcr.io/org-a</code> , which I am a member of.</p>
</blockquote>
</aside>
<p>Can you share the exact error?</p>
<p>Any org member can currently publish packages to the organization. However there is a setting for organizations that allow containers to be published at all.   See <a href="https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry/enabling-github-container-registry-for-your-organization">https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry/enabling-github-container-registry-for-your-organization</a></p>

**Comment by ataylorme** [comment]
<aside class="quote group-github-employees" data-username="clarkbw" data-post="6" data-topic="131217">
<div class="title">
<div class="quote-controls"></div>
 clarkbw:</div>
<blockquote>
<p>Can you share the exact error?</p>
</blockquote>
</aside>
<p>Sure, my terminal output is below (actual org name changed to org-a for privacy). I re-tagged the latest NodeJS container for my organization, logged out of <code>ghcr.io</code>, logged in with a personal access token that has the <code>write:packages</code> scope and attempted to push the container.</p>
<p>I am a member of the GitHub org but I am <strong>not</strong> an org owner.</p>
<pre><code class="lang-auto">ataylor ~ » docker tag node:latest ghcr.io/org-a/my-team-prefix/spark-node-test:0.0.1 
ataylor ~ » docker logout ghcr.io
Removing login credentials for ghcr.io
ataylor ~ » docker login ghcr.io
Username: ataylorme                                        
Password: 
Login Succeeded
ataylor ~ » docker push ghcr.io/org-a/my-team-prefix/spark-node-test:0.0.1 
The push refers to repository [ghcr.io/org-a/my-team-prefix/spark-node-test]
b9914fd725cf: Mounted from node:latest 
fb9a98332674: Mounted from node:latest 
c9b47a3c6380: Mounted from node:latest 
8211c12c1c23: Mounted from node:latest 
1d3ec06e3d4f: Mounted from node:latest 
9e5330403dba: Mounted from node:latest 
8bd20dc0b7e5: Mounted from node:latest 
94b70b410c2a: Mounted from node:latest 
3567db1eb737: Mounted from node:latest 
denied: permission_denied: create_package
</code></pre>

**Comment by clarkbw** [comment]
<aside class="quote no-group" data-username="ataylorme" data-post="7" data-topic="131217">
<div class="title">
<div class="quote-controls"></div>
 ataylorme:</div>
<blockquote>
<p>I am a member of the GitHub org but I am <strong>not</strong> an org owner.</p>
</blockquote>
</aside>
<p>Are you sure the organization has turned on the <a href="https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry/enabling-github-container-registry-for-your-organization">setting required for GHCR</a>?</p>

**Reply by mbaroody** [reply]
this link is now broken. is it still required to explicitly enable?

**Comment by ataylorme** [comment]
<aside class="quote group-github-employees" data-username="clarkbw" data-post="8" data-topic="131217">
<div class="title">
<div class="quote-controls"></div>
 clarkbw:</div>
<blockquote>
<p>Are you sure the organization has turned on the <a href="https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry/enabling-github-container-registry-for-your-organization" rel="noopener nofollow ugc">setting required for GHCR </a>?</p>
</blockquote>
</aside>
<p>That setting was not enabled. It is now (thanks for the link) and I am able to push a container to <code>ghcr.io</code>!</p>
<p><a class="mention" href="/clarkbw">@clarkbw</a> is there a way to have newly created containers on the GitHub container registry <code>read</code> by default? My goal is to have them be private within our organization but still allow any org member to use the container without having to be explicitly added each time we create a new container.</p>
<p>If there is not a way to do that by default is there an API I can use to grant teams or users access to containers published to <code>ghcr.io</code>?</p>

**Comment by clarkbw** [comment]
<aside class="quote no-group" data-username="ataylorme" data-post="9" data-topic="131217">
<div class="title">
<div class="quote-controls"></div>
 ataylorme:</div>
<blockquote>
<p>is there a way to have newly created containers on the GitHub container registry <code>read</code> by default?</p>
</blockquote>
</aside>
<p>We’re working on an “internal” visibility setting that be the new default for images.  Internal would mean all org members have read-access by default.</p>

**Comment by ataylorme** [comment]
<p>That is great news, thank you <a class="mention" href="/clarkbw">@clarkbw</a>!</p>

**Comment by alwell-kevin** [comment]
<p>Thanks for your support here!</p>

**Comment by sweco-nlecla** [comment]
<p>Hi,</p>
<p>I am having the same problems as mentioned above. When trying to push a package (which is a docker image) to our organisation. I get the error:<br>
<code>denied: permission_denied: create_package</code>.</p>
<p>When I publish it under my own account there is no problem, everything works fine.<br>
I am a member of the organisation but not an owner. The links mentioned in the thread are not available anymore. Is there still a setting which needs to be turned on in the organisation settings for enabling the creation of packages? Or is there something else which might have changed?</p>
<p>Thanks in advance.</p>

**Comment by sweco-nlecla** [comment]
<p>I fixed it by pushing it directly from a Github action using the GITHUB_TOKEN.</p>

**Comment by coherent-andyaaz** [comment]
<p>I had the same issue (trying to publish image as org member). you need to go to package permissions &gt; container creation in your org settings, and tick “internal”. But I cannot find any documentation about this requirement. I think this page should have mentioned it <a href="https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images" class="inline-onebox" rel="noopener nofollow ugc">Working with the Container registry - GitHub Docs</a></p>

**Comment by carlosgjs** [comment]
I'm hitting the same issue. I can push to my personal account, but not to my organization's. Internal package creation is already enabled:
<img width="878" alt="image" src="https://user-images.githubusercontent.com/10456546/227651804-f2c1cf2e-8d9c-43d3-ad64-edae219ea12f.png">

My token has the following permissions:
<img width="535" alt="image" src="https://user-images.githubusercontent.com/10456546/227651946-28c2d36f-b995-4a59-8793-2b6d0397d1e8.png">

But I'm still getting:
```
$  docker push ghcr.io/my_org/image-name:1.0
8d2c96ce8e1a: Layer already exists
10dbcddbc26a: Layer already exists
67b3a844367e: Layer already exists
f6a8e6388941: Layer already exists
eb0387e3d412: Layer already exists
bc01256d1a39: Layer already exists
3804935bde62: Layer already exists
denied: permission_denied: create_package
```
Any ideas?

**Reply by johnaroj** [reply]
+1

**Reply by andresv** [reply]
@carlosgjs make sure you are not Outside Collaborator in https://github.com/orgs/my_org/outside-collaborators.

**Reply by ernests** [reply]
Encountering the same problem. I'm owner of organization and repo. 
`denied: permission_denied: create_package`

So confusing

**Comment by ernests** [comment]
Followed @sweco-nlecla example and created Action for repo. It creates package
